### PR TITLE
move txn template filler above fee setting

### LINKED ISSUE #3591: Goal asset send does not respect fee set to 0
### Subject of the issue

When trying to create a group transaction where the fee on some transactions are set to 0 goal does not respect the 0 setting

```
ben@Algo-Guidarelli:~$ sb goal asset send --assetid 1 -f P75HT5DGUZ3TXI2TAESMJ2O7SMIC3WYFFQP5XARFTMAINI6OXOHANVMCOM -t P75HT5DGUZ3TXI2TAESMJ2O7SMIC3WYFFQP5XARFTMAINI6OXOHANVMCOM -a 100 -o tmp.txn --fee 0
ben@Algo-Guidarelli:~$ sb goal clerk inspect tmp.txn
tmp.txn[0]
{
  "txn": {
    "aamt": 100,
    "arcv": "P75HT5DGUZ3TXI2TAESMJ2O7SMIC3WYFFQP5XARFTMAINI6OXOHANVMCOM",
    "fee": 1000,
    "fv": 1,
    "gh": "XNWO3Gc3JDDO1Uj0pab/aHGs0RZAJAnwVwrnC4fWuxM=",
    "lv": 1001,
    "note": "MNg6BVaswnM=",
    "snd": "P75HT5DGUZ3TXI2TAESMJ2O7SMIC3WYFFQP5XARFTMAINI6OXOHANVMCOM",
    "type": "axfer",
    "xaid": 1
  }
}
```

### Your environment
sandbox with feature/contract-to-contract

### PR DESCRIPTION
**Pull request by barnjamin**
## Summary

closes #3591 

## Test Plan

```sh
goal asset send --assetid 1 -f P75HT5DGUZ3TXI2TAESMJ2O7SMIC3WYFFQP5XARFTMAINI6OXOHANVMCOM -t P75HT5DGUZ3TXI2TAESMJ2O7SMIC3WYFFQP5XARFTMAINI6OXOHANVMCOM -a 100 -o tmp.txn --fee 0

goal clerk inspect tmp.txn # should have no fee set
```

### CODE REVIEW

File: cmd/goal/asset.go [hunk 1]
```diff
@@ -545,6 +538,13 @@ var sendAssetCmd = &cobra.Command{
 		if err != nil {
 			reportErrorf("Cannot construct transaction: %s", err)
 		}
+		explicitFee := cmd.Flags().Changed("fee")
```
Comment (algochoi) [inline cmd/goal/asset.go:541]: Is this change okay for single transactions? I see that in `client.FillUnsignedTxTemplate(sender, firstValid, lastValid, fee, tx)`, there is an explicit check for [`if fee == 0`](https://github.com/algorand/go-algorand/blob/941c8204591b166e68bb205d04ab0eb0b7b318cb/libgoal/transactions.go#L402) to set the fee to the minimum Tx Fee. It looks like this change might override that setting for both single and group transactions.

**Review by algochoi — COMMENTED** [review]
Just had a question about setting min fees

**Review by algochoi — DISMISSED** [review]

**Comment by jannotti** [comment]
Could I convince you to return the note and lease processing back to their old spot?  It is correct in either location, but all the other commands do those lines early.  I'd rather keep the commands as similar to one another as possible (increasing the likelihood that some enthusiastic potential contributor abstracts the commonality and puts all that work in one place)

**Review by jannotti — APPROVED** [review]
Thanks for catching. I'm pretty sure that was my bug.

**Review by ahangsu — APPROVED** [review]
sorry for the late response, I overlooked the `FillUnsignedTxTemplate` and I was wondering what could be wrong. Now I am convinced.

**Pull request merged**